### PR TITLE
Add some background material about AMR to step-6.

### PR DESCRIPTION
**Pull request by bangerth**
This is not all I want to do for #6524, but I've learned that I need to be content
with making small progress to get to the bigger goals. It is almost midnight here,
after all, and I have another 6 hours to teach tomorrow.

I'll get back to adding more information to the remainder of the introduction
at a later point.

### CODE REVIEW

File: examples/step-6/doc/intro.dox [hunk 1]
```diff
@@ -3,12 +3,105 @@
...
+  edge to the opposite vertex. Of course, the midpoint from the longest edge
+  has to somehow be balanced by *also* refining the cell on the other side of
+  that edge (if there is one). If the edge in question is also the longest
+  edge of the neighboring cell, then we can just run a new edge through the
+  neighbor to the opposite vertex; otherwise a slightly more involved
+  construction is necessary that adds more new vertices on at least one
+  other edge of the neighboring cell, and then may propagate to the neighbors
+  of the neighbor until the algorithm terminates. This is hard to describe 
+  in words, and because deal.II does not use triangles not worth the time here.
+  But if you're curious, you can always watch video lecture 15 at the link
```
Comment (davydden) [inline examples/step-6/doc/intro.dox]: just give the link here, something like `<a href="http://blabla.com">here</a>` could work.

File: examples/step-6/doc/intro.dox [hunk 2]
```diff
@@ -3,12 +3,105 @@
...
+that they always generate a mesh where the lines where two cells touch
+are entire edges of both adjacent cells. With a bit of work, this strategy
+is readily adapted to three-dimensional meshes made from tetrahedra.
+
+Neither of these methods works for quadrilaterals in 2d and hexahedra in 3d,
+or at least not easily. The reason is that the transition elements created
+out of the quadrilateral neighbors of a quadrilateral cell that is to be refined
+would be triangles, and we don't want this. Consequently,
+the approach to adaptivity chosen in deal.II is to use grids in which
 neighboring cells may be refined a different number of times. This then
```
Comment (davydden) [inline examples/step-6/doc/intro.dox]: I would reword it to say that the difference is 1 level always. I know you say it later, but i think this place is more appropriate.

File: examples/step-6/doc/intro.dox [hunk 3]
```diff
@@ -3,12 +3,105 @@
...
+with barely a dozen lines of additional code. The program shows what
+these lines are, as well as another important ingredient of adaptive
+mesh refinement (AMR): a criterion that can be used to determine whether
+it is necessary to refine a cell because the error is large on it,
+whether the cell can be coarsened because the error is particularly
+small on it, or whether we should just leave the cell as it is. We
+will discuss all of these issues in the following.
+
+
+<h3> How adaptively refined meshes look like </h3>
```
Comment (drwells) [inline examples/step-6/doc/intro.dox]: Should be 'what adaptively refined meshes look like'

File: examples/step-6/doc/intro.dox [hunk 4]
```diff
@@ -3,12 +3,105 @@
 
 @dealiiVideoLecture{15,16,17,17.25,17.5,17.75}
 
-The main emphasis in this example is the handling of locally refined
-grids. The approach to adaptivity chosen in deal.II is to use grids in which
+This program is finally about one of the main features of deal.II:
+the use of adaptively (locally) refined meshes. The program is still
+based on step-4 and step-5, and as you will see, it does not actually
```
Comment (drwells) [inline examples/step-6/doc/intro.dox]: I believe the comma should be after `and`, i.e., the sentence should read `The program is still based on step-4 and step-5 and, as you will see, it does...` since `as you will see` is a commentary.

File: examples/step-6/doc/intro.dox [hunk 5]
```diff
@@ -3,12 +3,105 @@
...
+- Mark those cells that have large errors for refinement, mark those that have
+  particularly small errors for coarsening, and leave the rest alone;
+- Refine and coarsen the cells so marked to obtain a new mesh;
+- Repeat the steps above on the new mesh until the overall error is
+  sufficiently small.
+
+For reasons that are probably lost to history (maybe that these functions
+used to be implemented in FORTRAN, a language that does not care about
+whether something is spelled in lower or UPPER case letters, with programmers
+often choosing upper case letters habitually), the loop above is often
```
Comment (drwells) [inline examples/step-6/doc/intro.dox]: IIRC correctly a lot of IBM computers back then used 6 bit encodings that did not allow for lower case letters (good old BCD)

File: examples/step-6/doc/intro.dox [hunk 6]
```diff
@@ -3,12 +3,105 @@
...
+referenced in publications about mesh adaptivity as the 
+SOLVE-ESTIMATE-MARK-REFINE loop (with this spelling).
+
+Beyond this structure, however, there are a variety of ways to achieve
+this. Fundamentally, they differ in how exactly one generates one mesh
+from the previous one.
+
+If one were to use triangles (which deal.II does not do), then there are
```
Comment (drwells) [inline examples/step-6/doc/intro.dox]: :)

File: examples/step-6/doc/intro.dox [hunk 7]
```diff
@@ -3,12 +3,105 @@
...
+
+@image html hanging_nodes.png "A simple mesh with hanging nodes"
+
+A more complicated two-dimensional mesh would look like this (and is
+discussed in the "Results" section below):
+
+<img src="https://www.dealii.org/images/steps/developer/step_6_grid_5_ladutenko.svg"
+     alt="Fifth adaptively refined Ladutenko grid: the cells are clustered
+          along the inner circle."
+     width="300" height="300">
```
Comment (drwells) [inline examples/step-6/doc/intro.dox]: Does this show up centered on the page? If not, would you mind using the `onecolumn` CSS environment?

Thanks for copying the alt-text. Its a long term goal of mine to add more of those to improve our accessibility.

File: examples/step-6/doc/intro.dox (hunk 1 above)
Comment (bangerth) [reply examples/step-6/doc/intro.dox]: We have avoided hardcoding these URLs for the videos and instead used macros to expand them. The video in question here is referenced above, and I don't want to provide a separate URL here -- that can only become outdated.

File: examples/step-6/doc/intro.dox (hunk 5 above)
Comment (bangerth) [reply examples/step-6/doc/intro.dox]: Yes. I didn't add this but can if you want.

File: examples/step-6/doc/intro.dox (hunk 7 above)
Comment (bangerth) [reply examples/step-6/doc/intro.dox]: Yes, centered for me.

**Comment by bangerth** [comment]
I've made most of the requested changes and commented on the others. All in front of 70 students who were incidentally watching this on the big screen :-)

File: examples/step-6/doc/intro.dox (hunk 5 above)
Comment (drwells) [reply examples/step-6/doc/intro.dox]: Well, what you wrote is still correct, so its not a big deal.

Comment (drwells) [reply examples/step-6/doc/intro.dox]: (so we can keep it as-is)

**Comment by drwells** [comment]
Hello @bangerth's students, from Troy, NY! :)
